web/src/search/input/Suggestion.tsx

```diff
@@ -81,7 +86,10 @@ export function createSuggestion(item: GQL.SearchSuggestion): Suggestion | undef
         case 'Repository': {
             return {
                 type: SuggestionTypes.repo,
-                value: item.name,
+                // Add "regex start and end boundaries" to
+                // correctly scope additional suggestions
+                value: '^' + item.name + '$',
```

Comment:
This should probably also escape special chars like `.`
```suggestion
                value: '^' + escapeRegExp(item.name) + '$',
```